deeplake/util/merge.py

```diff
@@ -465,12 +478,30 @@ def merge_tensor_data(
     is_class_label = target_tensor.meta.htype == "class_label"
     if is_class_label:
         class_names = target_tensor.info.class_names
-    for index in new_indexes:
-        sample = target_tensor[index]
-        if is_class_label and class_names:
-            sample = convert_to_text(sample.numpy(), class_names, return_original=True)
-        original_tensor.append(sample)
-        original_id_tensor[-1] = target_id_tensor[index]
+        is_class_label = bool(class_names)
+    copy_links_only = False
+    if is_class_label:
+        links = original_tensor.meta.links
+        original_tensor.meta.links = {}
+        try:
+            for index in new_indexes:
```

Comment:
why don't we retrieve all in one call and then convert that list? Is it due to memory constraints? In that case we can do this in batches